src/test/java/com/google/sps/PlaceGuideServletTest.java

```diff
@@ -0,0 +1,615 @@
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.NAME_PROPERTY, placeGuide.getName());
+    placeGuideEntity.setProperty(
+        DatastorePlaceGuideRepository.AUDIO_KEY_PROPERTY, placeGuide.getAudioKey());
+    placeGuideEntity.setProperty(
+        DatastorePlaceGuideRepository.CREATOR_ID_PROPERTY, placeGuide.getCreatorId());
+    placeGuideEntity.setProperty(
+        DatastorePlaceGuideRepository.IS_PUBLIC_PROPERTY, placeGuide.isPublic());
+    placeGuideEntity.setProperty(
+        DatastorePlaceGuideRepository.PLACE_ID_PROPERTY, placeGuide.getPlaceId());
+    placeGuideEntity.setProperty(
+        DatastorePlaceGuideRepository.COORDINATE_PROPERTY, placeGuide.getCoordinate());
+    placeGuideEntity.setProperty(
+        DatastorePlaceGuideRepository.DESCRIPTION_PROPERTY, placeGuide.getDescription());
+    placeGuideEntity.setProperty(
+        DatastorePlaceGuideRepository.LENGTH_PROPERTY, placeGuide.getLength());
+    placeGuideEntity.setProperty(
+        DatastorePlaceGuideRepository.IMAGE_KEY_PROPERTY, placeGuide.getImageKey());
+    return placeGuideEntity;
+  }
+
+  private void saveUser(User user) {
+    datastore.put(getUserEntity(user));
+  }
+
+  private Entity getUserEntity(User user) {
+    Entity userEntity = new Entity(DatastoreUserRepository.ENTITY_KIND, user.getId());
+    userEntity.setProperty(DatastoreUserRepository.NAME_PROPERTY, user.getName());
+    userEntity.setProperty(DatastoreUserRepository.EMAIL_PROPERTY, user.getEmail());
+    userEntity.setProperty(
+        DatastoreUserRepository.BOOKMARKED_PLACE_GUIDES_IDS_PROPERTY,
+        user.getBookmarkedPlaceGuidesIds());
+    userEntity.setProperty(
+        DatastoreUserRepository.PUBLIC_PORTFOLIO_PROPERTY, user.portfolioIsPublic());
+    userEntity.setProperty(
+        DatastoreUserRepository.SELF_INTRODUCTION_PROPERTY, user.getSelfIntroduction());
+    userEntity.setProperty(DatastoreUserRepository.IMG_KEY_PROPERTY, user.getImgKey());
+    return userEntity;
+  }
+
+  /**
+   * This string is passed as an http request parameter. The format is the same as for
+   * LatLngBound.tourlValue() from the Maps Javascript API.
+   */
+  private String getRegionCornersString(GeoPt southWestCorner, GeoPt northEastCorner) {
+    return String.format(
+        "%f,%f,%f,%f",
+        southWestCorner.getLatitude(),
+        southWestCorner.getLongitude(),
+        northEastCorner.getLatitude(),
+        northEastCorner.getLongitude());
+  }
+
+  /** It sets the values of the request parameters. */
+  private void setupDoGetMockRequest(
+      PlaceGuideQueryType queryType, GeoPt southWestCorner, GeoPt northEastCorner)
+      throws IOException {
+    when(request.getParameter(PlaceGuideServlet.PLACE_GUIDE_QUERY_TYPE_PARAMETER))
+        .thenReturn(queryType.toString());
+    when(request.getParameter(PlaceGuideServlet.REGION_CORNERS_PARAMETER))
+        .thenReturn(getRegionCornersString(southWestCorner, northEastCorner));
+    sw = new StringWriter();
```

Comment:
there are not local, let's give them longer name

Reply:
Changed.